File: PP_02.01/DrivingSchoolWebAPI2/DrivingSchoolAPIModels/DbModels/InnerScheduleOfInstructor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DrivingSchoolAPIModels
{
    public class InnerScheduleOfInstructor
    {
        [Key]
        public int InnerScheduleOfInstructorId { get; set; }
        public int InstructorId { get; set; }
        [ForeignKey(nameof(InstructorId))]
        public Instructor Instructor { get; set; }
        public DateOnly DayOfWork { get; set; }
        public int? OuterScheduleId { get; set; }
        [ForeignKey(nameof(OuterScheduleId))]
        public OuterScheduleOfInstructor? OuterScheduleOfInstructor { get; set; }
        public bool IsOuterSchedule => OuterScheduleId != null;
    }
}

File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/InnerScheduleOfInstructor.kt
package com.example.drivingschoolappandroidclient.api.models

import com.example.drivingschoolappandroidclient.App
import java.sql.Date
import java.time.LocalDate

data class InnerScheduleOfInstructor(
var innerScheduleOfInstructorId : Int,
var instructorId : Int,
var instructor : Instructor,
var dayOfWorkJson : String,
var outerScheduleId : Int?,
var outerScheduleOfInstructor : OuterScheduleOfInstructor?,

){
    var dayOfWork : LocalDate
        get() = App.dateFromString(dayOfWorkJson)
        set(value) {
            dayOfWorkJson = App.dateToString(value)
        }
    val IsOuterSchedule = outerScheduleId != null
}
